### ISSUE TITLE
loss is negative

### ISSUE THREAD

**Issue by wujiachen987** [NVIDIA/semantic-segmentation#85]
I use your code train my own dataset, I imitate your `datasets/mapillary.py`  generate my datasets. Other is ok,but, the rmi_loss is
negative
 here is my logs
```
train.py --dataset staloader --cv 0 --result_dir logs/train_sta/cornflower-lynx_2020.10.21_10.53 --pre_size 256 --crop_size 256,256 --bs_trn 32 --lr_schedule poly --poly_exp 1.0 --optimizer sgd --lr 1e-2 --max_epoch 100 --rmi_loss --arch ocrnet.HRNet_Mscale --n_scales 0.5,1.0,2.0
None
Torch version: 1.4, 1.4.0
Using regular batch norm
n scales [0.5, 1.0, 2.0]
dataset = staloader
ignore_label = 255
num_classes = 8
all imgs 1000
all imgs 1000
Loading centroid file /code/semantic-segmentation-main/uniform_centroids/staloader_cv0_tile1024.json
Found 8 centroids
Class Uniform Percentage: 0.5
Class Uniform items per Epoch: 1000
cls 0 len 81556
cls 1 len 116605
cls 2 len 154647
cls 3 len 134469
cls 4 len 81019
cls 5 len 119233
cls 6 len 49323
cls 7 len 191353
Using Cross Entropy Loss
=> init weights from normal distribution
=>/code/semantic-segmentation-main/seg_weights/hrnetv2_w48_imagenet_pretrained.pth
Trunk: hrnetv2
Model params = 72.1M
Class Uniform Percentage: 0.5
Class Uniform items per Epoch: 1000
cls 0 len 81556
cls 1 len 116605
cls 2 len 154647
cls 3 len 134469
cls 4 len 81019
cls 5 len 119233
cls 6 len 49323
cls 7 len 191353
[epoch 0], [iter 1 / 31], [train main loss -1.581382], [lr 0.010000] [batchtime 0]
[epoch 0], [iter 2 / 31], [train main loss -2.413552], [lr 0.010000] [batchtime 0]
[epoch 0], [iter 3 / 31], [train main loss -2.152203], [lr 0.010000] [batchtime 0]
[epoch 0], [iter 4 / 31], [train main loss -2.387038], [lr 0.010000] [batchtime 0]
[epoch 0], [iter 5 / 31], [train main loss -2.925196], [lr 0.010000] [batchtime 0]
 ```
Can you give me any suggestion? Thanks.

**Comment by wujiachen987** [comment]
By the way, I use Cross Entropy is no problem.

**Comment by ajtao** [comment]
It's not an issue. See: https://github.com/ZJULearning/RMI/issues/2

**Comment by wujiachen987** [comment]
> It's not an issue. See: [ZJULearning/RMI#2](https://github.com/ZJULearning/RMI/issues/2)

Thank you for your reply. I will try use RMI loss in my dataset.
By the way, how do you judge whether the model fits
Is that the miou of the val set to judge the model? Loss metrics seems to have no value.

**Comment by bberry25** [comment]
Hi, I tried to train the model with my own data that has 2 classes.
Which values did you modify in the main scripts like config, train script, etc,.. 
Could you tell me the process of your transfer learning?

**Comment by wujiachen987** [comment]
I rewrite the dataset.py，config's path，not use script and acconding the error message to modify the code.

**Comment by bberry25** [comment]
> @bberry25
> I rewrite the dataset.py，config's path，not use script and acconding the error message to modify the code.

Thanks for your reply.
I have one more question about loading weights and pretrained model. 
Your dataset has only 7 classes and all pretrained  models have more than 7 classes from Cityscapes and Mapilaty dataset. Didn't you have an error from the differences in number of class between your own dataset and pretrained model? 
If there was an error, could you explain how did you solve it? 
Thank you so much
